internal/annotations/annotations.go

```diff
@@ -155,6 +155,36 @@ func ExtractKongPluginsFromAnnotations(anns map[string]string) []string {
 	return kongPluginCRs
 }
 
+type NamespacedKongPlugin struct {
+	Namespace string
+	Name      string
+}
+
+// ExtractNamespacedKongPluginsFromAnnotations extracts a KongPlugin name and optional namespace from an annotation
+// value. Plugins are delimited by ",". Values are either colon-delimited "namespace:name" strings or name-only
+// strings.
+func ExtractNamespacedKongPluginsFromAnnotations(anns map[string]string) []NamespacedKongPlugin {
+	var plugins []NamespacedKongPlugin
+	v := pluginsFromAnnotations(anns)
+	if v == "" {
+		return plugins
+	}
+	for _, s := range strings.Split(v, ",") {
```

Comment:
`plugins` could preallocate `len(strings.Split(v, ","))` entries

```suggestion
	v := pluginsFromAnnotations(anns)
	if v == "" {
		return nil
	}
	split := strings.Split(v, ",")
	plugins := make([]NamespacedKongPlugin, 0, len(split))

	for _, s := range split {
```